nestkernel/connector_model_impl.h

```diff
@@ -442,6 +442,89 @@ GenericConnectorModel< ConnectionT >::add_connection( Node& src,
   return conn;
 }
 
+/**
+ * Delete a connection of a given type directed to a defined target Node
+ * @param tgt Target node
+ * @param target_thread Thread of the target
+ * @param conn Connector Base from where the connection will be deleted
+ * @param syn_id Synapse type
+ * @return A new Connector, equal to the original but with an erased
+ * connection to the defined target.
+ */
+template < typename ConnectionT >
+ConnectorBase*
+GenericConnectorModel< ConnectionT >::delete_connection( Node& tgt,
+  size_t target_thread,
+  ConnectorBase* conn,
+  synindex syn_id )
+{
+  assert( conn != 0 ); // we should not delete not existing synapses
+  bool found = false;
+  vector_like< ConnectionT >* vc;
+
+  if ( conn->homogeneous_model() )
+  {
+    assert( conn->get_syn_id() == syn_id );
+    vc = static_cast< vector_like< ConnectionT >* >( conn );
+    // delete the first Connection corresponding to the target
+    for ( size_t i = 0; i < vc->size(); i++ )
+    {
+      ConnectionT* connection = &vc->operator[]( i );
```

Comment:
This looks funny... :smile: 
I only see the `operator[]` function used in to places: here and line 499. I guess, the proper way of using it would be `&( *vc )[ i ]`, which looks even more funny. Maybe we should go for the [`at`](http://www.cplusplus.com/reference/vector/vector/at/) function of vectors here: `&vc->at( i )`. If so, we should not use `assert` but throw an [`out_of_range`](http://www.cplusplus.com/reference/stdexcept/out_of_range/) exception.